### PR TITLE
Cleanup and extend user-directed output

### PR DESCRIPTION
**Pull request by tautschnig**
At various stages of the process CBMC can take a long time to progress. This PR adds additional output enabled at debug verbosity and cleans up some other user-directed output.

### CODE REVIEW

**Review by smowton — APPROVED** [review]

**Comment by tautschnig** [comment]
Would it be possible to set up a test run of TG for this, please? It should just impact output at debug level, but I have no insight as to whether such is being used anywhere, or at least being used in some tests.

**Review by martin-cs — APPROVED** [review]
Fine on my part.  I do wonder whether the ELOC changes should be part of it but ...

**Comment by tautschnig** [comment]
I have removed the additional statists (eloc related changes), they indeed did not fit this PR.

File: src/goto-programs/remove_function_pointers.cpp [hunk 1]
```diff
@@ -430,6 +430,26 @@ void remove_function_pointerst::remove_function_pointer(
   statistics().source_location=target->source_location;
   statistics() << "replacing function pointer by "
                << functions.size() << " possible targets" << eom;
+
+  // list the names of functions when verbosity is at debug level
+  if(get_message_handler().get_verbosity() >= M_DEBUG)
```
Comment (peterschrammel) [inline src/goto-programs/remove_function_pointers.cpp]: I suggest to add a function to the message handler that executes a function when the debug level check succeeds. You could then call, e.g. `debug(print_functions)` where `print_functions` could just be a lambda when it is short.

Comment (tautschnig) [reply src/goto-programs/remove_function_pointers.cpp]: Done: you can now do `conditional_output(debug(), <lambda>)`

**Review by peterschrammel — APPROVED** [review]
Definitely better than the previous solution!

**Pull request merged**